bin/client/src/witness.rs

```diff
@@ -44,7 +44,7 @@ impl<T: BlobProvider + Send> BlobProvider for BlobWitnessProvider<T> {
     ) -> Result<Vec<Box<Blob>>, Self::Error> {
         let blobs = self.provider.get_blobs(block_ref, blob_hashes).await?;
         for blob in &blobs {
-            let c_kzg_blob = c_kzg::Blob::from_bytes(blob.as_slice()).unwrap();
+            let c_kzg_blob = c_kzg::Blob::new(***blob);
```

Comment:
ah you might be right!
```
#[repr(C)]
#[derive(Debug, Hash, PartialEq, Eq)]
pub struct Blob {
    bytes: [u8; 131072usize],
}
```
Could the fact that its C representation mean its actually moved and not copied? If not, then you're right and I will just close this PR.

Reply:
Feel like c-kzg should have a move constructor though...!

Reply:
The memory representation doesn't affect the borrow checker afaik

Reply:
After reading a bit more, pretty convinced also this is true. But also don't think the borrow checker and the fact that a struct implements the Copy trait actually means it will be copied when passed into a function that doesn't modify it, even though rust passes arguments by value. And that's because of https://en.wikipedia.org/wiki/Copy_elision. Hence I do think the change in this PR *could* help (probably only when building with --release to have compiler optimizations). Nonetheless, this change would make a lot more sense if I implemented it uniformly across the codebase, *and* wrote some benchmark to actually show that it helps. So I'll close for now. Enjoy your holiday break! :)